internal/engine/compiler/impl_amd64_test.go

```diff
@@ -441,6 +441,61 @@ func TestAmd64Compiler_preventCrossedTargetdRegisters(t *testing.T) {
 	}
 }
 
+// Relates to #1111: older AMD64 CPUs do not support the LZCNT instruction
+// CPUID should be used instead. We simulate presence/absence of the feature
+// by overriding the field in the corresponding struct.
+func TestAmd64Compiler_ensureClz_ABM(t *testing.T) {
+	tests := []struct {
+		name        string
+		cpuFeatures cpuFeatureFlags
+		expectedLen int
+	}{
+		{
+			name:        "with ABM",
+			expectedLen: 140,
+			cpuFeatures: cpuFeatureFlags{
+				flags:      0,
+				extraFlags: cpuExtraFeatureABM,
+			},
+		},
+		{
+			name:        "without ABM",
+			expectedLen: 156,
+			cpuFeatures: cpuFeatureFlags{
+				flags:      0,
+				extraFlags: 0, // no flags, thus no ABM, i.e. no LZCNT
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			env := newCompilerEnvironment()
+
+			newCompiler := func() compiler {
+				c := newCompiler().(*amd64Compiler)
+				// override auto-detected CPU features with the test case
+				c.cpuFeatures = tt.cpuFeatures
+				return c
+			}
+
+			compiler := env.requireNewCompiler(t, newCompiler, nil)
+			err := compiler.compilePreamble()
+			require.NoError(t, err)
```

Comment:
I don' think these two are needed for this test.

Reply:
I tried already, but it looks like the state is inconsistent without them.

without the preamble:

```
panic: runtime error: index out of range [18446744073709551615] with length 0 [recovered]
...
github.com/tetratelabs/wazero/internal/engine/compiler.(*runtimeValueLocationStack).pop(...)
```

Reply:
so this I solve by pushing a constant i32